crates/oxc_semantic/src/builder.rs

```diff
@@ -351,12 +359,12 @@ impl<'a> SemanticBuilder<'a> {
     /// Declare an unresolved reference in the current scope.
     ///
     /// # Panics
-    pub fn declare_reference(&mut self, name: &str, reference: Reference) -> ReferenceId {
+    pub fn declare_reference(&mut self, name: &Atom<'a>, reference: Reference) -> ReferenceId {
```

Comment:
A small thing: `Atom<'a>` is just a wrapper around `&'a str`. So it's better to pass around `Atom`s, rather than `&Atom`s (`&Atom` is equivalent to `&&str` - double-reference). `atom.clone()` is very cheap (compiler treats it as if `Atom` was `Copy`).

We could make `Atom` `Copy` now, but I think we should not, to avoid breakage if we later make `Atom` an interned string.